tests/conftest.py

```diff
@@ -631,3 +631,12 @@ def make_404(request):
         text=station_listing,
         status_code=200,
     )
+
+
+def records_contain_db_connection(test_session, caplog):
+    for record in caplog.records:
+        if "database" in record.__dict__:
```

Comment:
I could be wrong here, but I believe that you can just use the idiom `if database in record` and do not have to specifically access the `__dict__` attribute. See: https://docs.python.org/3/reference/expressions.html#membership-test-operations

One thing that you *don't* check here, that maybe you should is whether the record is found in the correct level of logging. E.g. you check that a log entry is found, but it's possible that it's at a higher level of logging then specified.